nav2_behavior_tree/include/nav2_behavior_tree/plugins/action/controller_cancel_node.hpp

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2022 Pradheep Padmanabhan - Neobotix GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__PLUGINS__ACTION__CONTROLLER_CANCEL_NODE_HPP_
+#define NAV2_BEHAVIOR_TREE__PLUGINS__ACTION__CONTROLLER_CANCEL_NODE_HPP_
+
+#include <memory>
+#include <string>
+
+#include "nav2_behavior_tree/bt_service_node.hpp"
+
+#include "lifecycle_msgs/srv/change_state.hpp"
+
+namespace nav2_behavior_tree
+{
+
+/**
+ * @brief The ControllerCancel behavior is used for the transtition of the controller
+ * to the shutdown lifecycle state.
+ */
+class ControllerCancel : public BtServiceNode<lifecycle_msgs::srv::ChangeState>
```

Comment:
There is actually no ports. Just a service. Got inspired from [reinitialization_global_localization_service](https://github.com/ros-planning/navigation2/blob/main/nav2_behavior_tree/plugins/action/reinitialize_global_localization_service.cpp)

Reply:
Adding on, this is already done in the [BTServiceNoce](https://github.com/ros-planning/navigation2/blob/b201d058a7b15f1786c57afe69d079ac5d502018/nav2_behavior_tree/include/nav2_behavior_tree/bt_service_node.hpp#L93)

Reply:
OK, docs need to show this though since its part of this node, even if its not in this implementation's code (since its in the derived path to be relevant)